app/ota_client_stub.py

```diff
@@ -162,32 +182,57 @@ def host_addr(self):
 
     async def update(self, request: v2.UpdateRequest) -> v2.UpdateResponse:
         logger.info(f"{request=}")
+        my_ecu_id = self._ecu_info.get_ecu_id()
+        _loop = asyncio.get_running_loop()
+
         response = v2.UpdateResponse()
+        # TODO: add a method for ota_client to proxy request_update
+        if not self._ota_client.live_ota_status.request_update():
+            # current ota_client's status indicates that
+            # the ota_client should not start an ota update
+            logger.warning(
+                f"ota_client should not take ota update under ota_status={self._ota_client.live_ota_status.get_ota_status()}"
+            )
+            ecu = response.ecu.add()
+            ecu.ecu_id = my_ecu_id
+            ecu.result = v2.RECOVERABLE
 
-        # init state machine(P1: ota_service, P2: ota_client)
-        ota_sfsm = OtaStateSync()
-        ota_sfsm.start(caller=ota_sfsm._P1_ota_service)
+            return response
+        else:
+            # update request permitted, prepare response
+            ecu = response.ecu.add()
+            ecu.ecu_id = my_ecu_id
+            ecu.result = v2.NO_FAILURE
```

Comment:
In case of my_ecu_id is not listed in the request, my_ecu_id should not be added in the response. We REQUIRED this feature.

Reply:
Sure! Let me re-think about the logic here.